astropy/modeling/separable.py

```diff
@@ -301,7 +301,10 @@ def _separable(transform):
         An array of shape (transform.n_outputs,) of boolean type
         Each element represents the separablity of the corresponding output.
     """
-    if isinstance(transform, CompoundModel):
+    transform_matrix = transform._calculate_separability_matrix()
+    if transform_matrix is not NotImplemented:
+        return transform_matrix
```

Comment:
```suggestion
    if transform_matrix := transform._calculate_separability_matrix() is not NotImplemented:
        return transform_matrix
```

Reply:
oh I forgot we can do this now :heart_eyes_cat: